First PR! added reltol and abstol to rank(A) in linalg/generic.jl

miguelraz opened the pull request:
Just tried adding some basic keyword functionality to the funciton.

miguelraz committed:
Adds reltol and abstol cases to rank() function

miguelraz committed:
Merge branch 'master' of https://github.com/JuliaLang/julia into rank_tolerance_fix

StefanKarpinski commented:
This breaks tests and compatibility since the old form `rank(M, tol)` doesn't work. In general, when you submit a PR, you should try running the tests (at least the relevant ones) locally first. In this case, that means `make test-linalg` since the `rank` function appears in various `test/linalg` files.

It's also probably easier to get some buy-in for and API change like this before submitting a PR.

miguelraz commented:
@StefanKarpinski Thanks a lot for the feedback. I have just noticed something with my notifications and will try to be more attentive to the thread.

I didn't run tests because I just assumed that if merge with master was compatible, all was good.

Will run tests and adjust accordingly.

If you have any other recommendations that are worth hearing, I'm all for it.

See ya around, and thanks!

miguelraz committed:
WIP/ RFC/First PR: Add `unzip` to base.iterators.jl

~~@StefanKarpinski~~ Julia needs an underling to give this PR a shot. [#13942](https://github.com/JuliaLang/julia/issues/13942).
1. I know I'm clearly missing some important things, but I hope the effort counts. I couldn't find a way to get the rows from the array comprehension...
2. Willing to start over from scratch - any comments/critiques/suggestions more than welcome.
3. I am first trying a MWE with simple structure and inputs. Then I will try to get something that works more generically. Is this a good way to procceed?

[ci skip]

Roadmap:
* Verify each returned array is the same type as the types of the elements
* Verify lengths of all arrays to be unzipped
* Return as a single Array of Arrays or as separate Arrays?
* Remove that `zip is its own inverse` in `@doc zip`?
* Revise documentation phrasing and example

miguelraz committed:
WIP/ RFC/First PR: Add `unzip` to base.iterators.jl

~~@StefanKarpinski~~ Julia needs an underling to give this PR a shot. [#13942](https://github.com/JuliaLang/julia/issues/13942).
1. I know I'm clearly missing some important things, but I hope the effort counts. I couldn't find a way to get the rows from the array comprehension...
2. Willing to start over from scratch - any comments/critiques/suggestions more than welcome.
3. I am first trying a MWE with simple structure and inputs. Then I will try to get something that works more generically. Is this a good way to procceed?

[ci skip]

Roadmap:
* Verify each returned array is the same type as the types of the elements
* Verify lengths of all arrays to be unzipped
* Return as a single Array of Arrays or as separate Arrays?
* Remove that `zip is its own inverse` in `@doc zip`?
* Revise documentation phrasing and example

miguelraz committed:
Merge pull request #1 from JuliaLang/master

update julia fork

miguelraz committed:
Merge remote-tracking branch 'miguelraz/patch-3'

miguelraz committed:
adding tests to unzip

miguelraz committed:
Merge branch 'master' into rank_tolerance_fix

miguelraz commented:
This PR got too mucky. Will attempt again with a new one.
